include/igl/generate_sparse_marching_cubes_grid.cpp

```diff
@@ -0,0 +1,152 @@
+#include "generate_sparse_marching_cubes_grid.h"
+
+#include <unordered_map>
+#include <array>
+#include <vector>
+
+
+struct RowVector3iHash {
```

Comment:
See the style-guidelines. Every .cpp should have a single function and all new functions should be in the igl namespace